python/lsst/rapid/analysis/quickLook.py

```diff
@@ -0,0 +1,281 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from lsst.ip.isr import IsrTask
+from lsst.ip.isr.isrTask import IsrTaskConnections
+import lsst.pipe.base.connectionTypes as cT
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask
+from lsst.meas.algorithms.installGaussianPsf import InstallGaussianPsfTask
+
+__all__ = ['QuickLookTask', 'QuickLookTaskConfig']
+
+
+def _getArgs(connection):
+    """Get the all the required args from a connection in order to reconstruct.
+    """
+    newArgs = {}
+    for attr in ("name", "storageClass", "multiple", "doc", "dimensions", "isCalibration", "deferLoad",
+                 "minimum", "lookupFunction"):
+        if hasattr(connection, attr):
+            newArgs[attr] = getattr(connection, attr)
+    return newArgs
+
+
+class QuickLookTaskConnections(IsrTaskConnections):
+    def __init__(self, *, config=None):
+        """Copy isrTask's connections, changing prereq min values to zero.
+
+        Copy all the connections directly for IsrTask, keeping ccdExposure as
+        required as non-zero, but changing all the other PrerequisiteInputs'
+        minimum values to zero.
+        """
+        super().__init__(config=IsrTask.ConfigClass())  # need a dummy config, isn't used other than for ctor
+        for name, connection in self.allConnections.items():
+            if hasattr(connection, 'minimum'):
+                args = _getArgs(connection)
+                if name != "ccdExposure":  # need one input image always
+                    args['minimum'] = 0
+                newConnection = type(connection)(**args)
+                self.allConnections[name] = newConnection
+                setattr(self, name, newConnection)
+
+        exposure = cT.Output(  # called just "exposure" to mimic isrTask's return struct
+            name="quickLookExp",
+            doc="The quickLook output exposure.",
+            storageClass="ExposureF",
+            dimensions=("instrument", "exposure", "detector"),
+        )
+        self.outputExposure = exposure
```

Comment:
I think because it's making it clear that it's a direct copy of the `exposure` just defined above. I could copy & paste that whole definition over, but that seems worse, and makes it less clear that they must be identical I think. But I'll admit I'm now unsure as to why it was ever like that. I should certainly have at least left a comment.

Reply:
Did you sort out why you did it this way?

Reply:
Yup, added comments for everything now.